### DISCUSSION TITLE
Do the "strict" and "strictQuery" options apply to subdocuments not in schema?

### DISCUSSION THREAD

**Discussion by rwmia** [Automattic/mongoose#11632] [Q&A]
I have a question about strict and strictQuery, specifically how they apply to subdocuments in Mongoose 6.x

Lets say I have this schema below:
```
var zooAnimalSchema = new Schema({
  animal_name: {
    type: String
   },
  animal_features: {
    trunk_length: {
      type: String
    }
  }
},
{
strict: true,
strictQuery: true
});
```

**Example 1**
```
zooAnimalSchema.findOneAndUpdate({
        _id: animal_obj_id
      }, {
        $set: {
         animal_features: {
             trunk_length: '40',
             random_property: 'random text'
          }
        }
      })
      .exec(function (err, res) {});
```

Question: Will "random_property" get saved to the doc even though it is not in the schema?

**Example 2**
```
zooAnimalSchema.find({
        'animal_features.random_property' : 'random_text'
      })
      .exec(function (err, res) {});
```

Question: Will "animal_features.random_property" get queried even though it is not in the schema?

**Comment by rwmia** [comment]
Can anyone with knowledge please answer? I am worried about upgrading mongoose due to the strict and strictQuery, specifically if applied to subdocuments.

**Comment by vkarpov15** [comment]
@rwmia `random_property: 'random text'` will **not** get saved if `strict: true`.

`strictQuery` only affects the filter portion of the query, `_id: animal_obj_id` in your example.
